aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -1087,7 +1091,9 @@ bool TensorIterator::fast_set_up(const TensorIteratorConfig& config) {
           auto& op = operands_[i];
           if (!op.tensor.defined()) {
             TORCH_INTERNAL_ASSERT(op.is_type_defined(), "no type for operand", i);
-            op.tensor = at::empty(shape_, op.options(), MemoryFormat::Contiguous);
+            AutoNonVariableTypeMode non_variable_type_mode;
+            const DeviceGuard device_guard(op.options().device());
```

Comment:
NB: DeviceGuard is not needed here, it's needed only for scalar-tensor case in non-fast setup, remove it later.